src/middlewares/asyncWrapper.js

```diff
@@ -5,12 +5,15 @@
   * @function
   *
   * @param {Function} wrappedFunction - the main controller
+  * @param {Boolean} middleware - A flag to know if it is a middleware
   *
   * @return {Function} - callback that execute the controller
   */
-export default wrappedFunction => async (request, response, next) => {
+export default (wrappedFunction, middleware = false) => async (request, response, next) => {
   try {
-    const { status, data, message } = await wrappedFunction(request, response, next);
+    const { status, data, message } = await wrappedFunction(request, response, next) || {};
```

Comment:
Why do we need the fallback object?

Reply:
The reason for the empty object is because the asyncWrapper was throwing an error that it cannot destructure undefined. 

This is basically for middlewares that do not return anything. We are just trying to make the asyncWrapper far more robust so that it can be used for middlewares that call the `next()` or that `return { status, data, message }`

Reply:
The asyncWrapper was refactored to make it far more robust. When used for a middleware that calls `next()` it fails because it is expecting an object to be returned to destructure `{status, data, message}`.

Without putting the empty object, I get an error e.g `cannot destructure status from undefined`

Reply:
This pattern seems weird. I'll recheck. Middleware should either return the response or return next(). I'm wondering why some middlewares return an object.